src/utils/index.ts

```diff
@@ -77,6 +77,20 @@ const conflictsSeverity = (first: SlotInfo, second: SlotInfo): number => {
   return (isMandatory(first) && isMandatory(second)) ? 2 : 1;
 }
 
+const classesConflictSeverity = (first: ClassInfo, second: ClassInfo): number => {
+  let maxSeverity = 0;
+
+  for (const slot of first.slots) {
+    for (const otherSlot of second.slots) {
+      if (schedulesConflict(slot, otherSlot)) {
+        maxSeverity = Math.max(maxSeverity, conflictsSeverity(slot, otherSlot));
+      }
+    }
+  }
+
+  return maxSeverity;
+}
```

Comment:
```suggestion
const classesConflictSeverity = (first: ClassInfo, second: ClassInfo): number =>
  first.slots.flatMap(slot =>
    second.slots
      .filter(otherSlot => schedulesConflict(slot, otherSlot))
      .map(otherSlot => conflictsSeverity(slot, otherSlot))
  ).reduce((maxSeverity, severity) => Math.max(maxSeverity, severity), 0);
```

Because who needs friends when you have pure functions?

Reply:
I understand that the clone has less lines but it becomes less readable and harder to debug because if we wanted to console log something inside one of the functions it would be a lot harder.

It is also harder because the majority of the people doesn't know well the workings of `reduce`.

Reply:
That's why we should have PFL in the first year, ahaha